File: Week_04/69/sqrtx.go
package main

func mySqrt(x int) int {
	left, right := 0, x

	for left <= right {
		mid := (left + right) / 2
		square := mid * mid
		if square == x {
			return mid
		} else if square > x {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return right
}

File: Week_04/69/sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        while left <= right:
            mid = (left + right) // 2
            square = mid * mid
            if square == x:
                return mid
            elif square > x:
                right = mid - 1
            else:
                left = mid + 1
        return right
